### PR TITLE
[BACKLOG-7786] Report execution cancelled when dashboard with unsaved…

### PR DESCRIPTION
**Pull request by Aliaksandr-Kastenka**
… changes is closed

Goes with https://github.com/pentaho/pentaho-platform/pull/2892
@tmorgner 
Please review.

### CODE REVIEW

File: pentaho-gwt-widgets/src/org/pentaho/gwt/widgets/client/tabs/PentahoTab.java [hunk 1]
```diff
@@ -148,10 +148,6 @@ public String getLabelTooltip() {
   }
 
   protected void closeTab() {
-    if ( getContent().getElement().getElementsByTagName( "iframe" ).getLength() > 0 ) {
```
Comment (tmorgner) [inline pentaho-gwt-widgets/src/org/pentaho/gwt/widgets/client/tabs/PentahoTab.java]: Why do you remove that now? We still want to know when a tab is closed to issue the cancel request.

File: pentaho-gwt-widgets/src/org/pentaho/mantle/client/events/EventBusUtil.java [hunk 2]
```diff
@@ -21,11 +21,12 @@
 import com.google.gwt.core.client.JavaScriptObject;
 import com.google.gwt.event.shared.EventBus;
 import com.google.gwt.event.shared.SimpleEventBus;
+import com.google.gwt.event.shared.HandlerRegistration;
 
 public interface EventBusUtil {
   public static final EventBus EVENT_BUS = GWT.create( SimpleEventBus.class );
 
-  public void addHandler( String eventType, JavaScriptObject handler );
+  HandlerRegistration addHandler( String eventType, JavaScriptObject handler );
```
Comment (tmorgner) [inline pentaho-gwt-widgets/src/org/pentaho/mantle/client/events/EventBusUtil.java]: Rebase the EWOK branch against master and you get that change in - we needed to add it during the merge as the user-console changes did not compile without it.

File: pentaho-gwt-widgets/src/org/pentaho/gwt/widgets/client/tabs/PentahoTab.java (hunk 1 above)
Comment (Aliaksandr-Kastenka) [inline pentaho-gwt-widgets/src/org/pentaho/gwt/widgets/client/tabs/PentahoTab.java]: I've add this event to MantleTabPanel:
https://github.com/pentaho/pentaho-platform/pull/2892

Comment (tmorgner) [inline pentaho-gwt-widgets/src/org/pentaho/gwt/widgets/client/tabs/PentahoTab.java]: In that case, then move the fireCloseTab(..) method as well. If that method is only used from within the MantleTabPanel, then make it part of that class - it then has no reason to exist here.

**Comment by Aliaksandr-Kastenka** [comment]
@tmorgner 
Pull-request updated.

**Pull request closed without merging**